### PR TITLE
fixed s_vals divergence across GPUs

### PR DESCRIPTION
**Pull request by Mustafa-cuda-dev**
s_vals were different on different GPUs (A100, H100, RTX 4090) with same inputs.

Fixed it by using fixed-point quantization + FP64 for accumulation. Now they match exactly.

Tested on A100, H100, RTX 4090 — all give same s_vals now.

Performance: ~35ms on A100, ~14.5ms on H100, ~52ms on RTX 4090.
Memory: ~19GB with FP16 weights — fits on RTX 4090.

Works with 7B and 13B models.

## Summary by CodeRabbit

* **New Features**
  * Added configuration options for proof quantization and deterministic processing.
  * Added a deterministic projection path to improve reproducibility.
* **Bug Fixes**
  * Improved recovery from GPU memory exhaustion during proof computation.
  * Reduced peak memory usage when processing logits and large hidden dimensions.
* **Compatibility**
  * Preserved existing proof computation interfaces and return formats.

### CODE REVIEW

File: grail/environments/proofs.py [hunk 1]
```diff
@@ -261,8 +294,57 @@ def compute_proofs(
...
+                has_projected_arg = False
+
+            if has_projected_arg:
+                commitments = verifier.create_commitments_batch(h_layer, r_vec, projected_s_vals=s_vals)
+            else:
+                raise RuntimeError(
+                    "Verifier must support 'projected_s_vals' argument. Please update GRAILVerifier."
+                )
+        else:
+            commitments = verifier.create_commitments_batch(h_layer, r_vec)
```
[the remark this answers is not in the document]
Comment (Mustafa-cuda-dev) [reply grail/environments/proofs.py:347]: Fixed in #70 — added projected_s_vals parameter to verifier.
